client/src/main/java/io/pravega/client/state/impl/RevisionedStreamClientImpl.java

```diff
@@ -142,6 +142,9 @@ public void writeUnconditionally(T value) {
             if (startOffset < segmentInfo.getStartingOffset()) {
                 throw new TruncatedDataException(format("Data at the supplied revision {%s} has been truncated. The current segment info is {%s}", start, segmentInfo));
             }
+            if (startOffset == endOffset) {
+                log.info("No new updates to be read from revision {}", start);
```

Comment:
Please include more identifying information (segment name, offsets, etc - anything that may be useful in debugging).

Also, do these need to be INFOs - how often are we going to log them?

Reply:
The start object of type Revision logs all the details like segment name and offsets. I have changed the log level to DEBUG since INFO is quite noisy is a normal run.